eslint-plugin-expensify/tests/no-api-side-effect-actions.test.js

```diff
@@ -0,0 +1,86 @@
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../no-api-side-effect-actions');
+const message = require('../CONST').MESSAGE.NO_API_SIDE_EFFECTS_ACTIONS;
+
+const ruleTester = new RuleTester({
+    parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module',
+    },
+});
+
+ruleTester.run('no-api-side-effect-actions', rule, {
+    valid: [
+        {
+            code: `
+                function test() {
+                    API.write('Report_AddComment', params);
+                }
+            `,
+            filename: './src/libs/actions/Test.js',
+        },
+        {
+            code: `
+                import Action from './src/libs/actions/Action.js'
+                function test() {
+                    Action.perform(params);
+                }
+            `,
+            filename: './src/libs/actions/Test.js',
```

Comment:
In this example we are importing an action and wrapping it in a function. Maybe this should not be allowed as you should call the action directly not wrap it in another action 🤔

Reply:
I am not sure if I understand what you mean here. Do you mean we should have `Action()`? I think we currently get the actions in such a way but I agree that will probably be refactored soon, for example of [Inbox.requestInboxCall()](https://github.com/Expensify/App/blob/51fb61808d84f4aae4c6cbb3f6875d43d355fbb5/src/pages/RequestCallPage.js#L153).

If we will have strictly one action file - one method, then I think the rule you suggest would make a lot of sense. Did I understand it right?

Reply:
**clarification**: the example in the test case is not something anyone would ever do - yet implied as "valid"
**suggestion:** we should remove it unless we can think of a real world example where someone would need to create a function that wraps an action.

Reply:
To confirm we're talking about this example yes?
```JS
        {
            // Make sure that we will not test a file which is not in '/actions/'
            code: `
                import Action from './src/libs/actions/Action.js'
                function test() {
                    API.write('Report_AddComment', params);
                    Action(params);
                }
            `,
            filename: './src/libs/notActions/Test.js',
        },
```

Yes, that should be invalid.

Reply:
Good yeah, that one is valid, because it is not in the `/actions/` directory

But we have an invalid which is as you described so I guess that is good.
```
{
            // Making an API call and calling an Action should be invalid
            code: `
                import Action from './src/libs/actions/Action.js'
                function test() {
                    API.write('Report_AddComment', params);
                    Action(params);
                }
            `,
            errors: [{
                message,
            }],
            filename: './src/libs/actions/Test.js',
        },
```

Reply:
Ah okay, thanks for clarifying!